sgkit/stats/association.py

```diff
@@ -209,9 +209,15 @@ def gwas_linear_regression(
 
     G = _get_loop_covariates(ds, dosage=dosage, call_genotype=call_genotype)
 
-    X = da.asarray(concat_2d(ds[list(covariates)], dims=("samples", "covariates")))
-    if add_intercept:
-        X = da.concatenate([da.ones((X.shape[0], 1), dtype=X.dtype), X], axis=1)
+    if len(covariates) == 0:
+        if add_intercept:
+            X = da.ones((ds.dims["samples"], 1), dtype=float)
```

Comment:
Any objection to float32 here to avoid promotion if `G` is 32-bit as well?

Reply:
Good call